R/human_infection.R

```diff
@@ -309,8 +309,6 @@ schedule_infections <- function(
       infection_times
     )
     events$detection$schedule(to_infect, infection_times)
-    events$subpatent_infection$clear_schedule(to_infect)
```

Comment:
On infection, we would cancel infected individuals progressing to their next state.

Initially this makes sense. If you are infecting from U -> D, the individual is already scheduled to go to S and if you don't cancel it, your infection won't last!

But this is the cause of the competing hazard. By cancelling subpatent_infection and recovery (A -> U -> S) we reduce the rates `da` and `du`.

Instead, for recovery, we want to sample individuals from U who are NOT going to be reinfected.